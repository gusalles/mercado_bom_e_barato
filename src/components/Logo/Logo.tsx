import Image from 'next/image';

interface LogoProps {
  width?: number;
  height?: number;
}

export function Logo({ width = 145, height = 40 }: LogoProps) {
  return (
    <Image
      src="/logo.svg"
      alt="Logo Mercado B&B"
      width={width}
      height={height}
    />
  );
}
