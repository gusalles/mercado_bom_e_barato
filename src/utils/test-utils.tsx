import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

export * from '@testing-library/react';

export function renderWithClient(ui: ReactElement) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
  );
}
