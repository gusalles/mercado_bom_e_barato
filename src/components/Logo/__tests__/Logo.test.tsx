import { render, screen } from '@/utils/test-utils';
import { Logo } from '../Logo';

describe('Logo', () => {
  it('should render with default dimensions', () => {
    render(<Logo />);

    const logo = screen.getByRole('img');

    expect(logo).toBeInTheDocument();
    expect(logo).toMatchSnapshot();
  });

  it('should render with custom dimensions', () => {
    render(<Logo width={200} height={60} />);

    const logo: { width: number } & HTMLElement = screen.getByRole('img');

    expect(logo).toBeInTheDocument();
    expect(logo.width).toEqual(200);
  });
});
