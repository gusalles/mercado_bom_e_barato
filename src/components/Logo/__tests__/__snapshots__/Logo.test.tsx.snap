// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > should render with default dimensions 1`] = `
<img
  alt="Logo Mercado B&B"
  data-nimg="1"
  decoding="async"
  height="40"
  src="/logo.svg"
  style="color: transparent;"
  width="145"
/>
`;
